core/src/main/scala/kafka/consumer/ConsumerFetcherManager.scala

```diff
@@ -69,7 +69,7 @@ class ConsumerFetcherManager(private val consumerIdString: String,
                                                             config.clientId,
                                                             config.socketTimeoutMs,
                                                             correlationId.getAndIncrement).topicsMetadata
-        if(logger.isDebugEnabled) topicsMetadata.foreach(topicMetadata => debug(topicMetadata.toString()))
+        if(isDebugEnabled) topicsMetadata.foreach(topicMetadata => debug(topicMetadata.toString()))
```

Comment:
I think it might be useful to use macros at this place (and some other similars below). Any ideas pro/con from reviewers?

Reply:
With `scalalogging`, we don't need to do `isDebugEnabled` checks if we use the `logger` instead of the methods with `by name` parameters.

Reply:
Yea, I left this check here because I thought there were optimization reasons behind putting an if check for this for loop, so we don't do an empty loop. Although it you think it's an unnecessary optimization, I can get rid of it. :)
I also thought about doing a macro for this similarly as scala-logging does but then I discarded this as I wasn't sure about what gains I'd get.

Reply:
Oh, it's a loop. We can probably just leave it, this code will be removed in 2.0.0.